fake_dependencies/harald/server.js

```diff
@@ -0,0 +1,235 @@
+const express = require("express");
+const fs = require("fs");
+const util = require("util");
+const uuidv1 = require('uuid/v1');
+
+const port = process.env.port || 3000;
+
+const app = express();
+app.use(express.json());
+
+const readFile = util.promisify(fs.readFile);
+const writeFile = util.promisify(fs.writeFile);
+const serialize = (data) => JSON.stringify(data, null, 2);
+const deserialize = (text) => JSON.parse(text);
+
+app.get("/api/v1/channels", (req, res) => {
+    readFile("./channels-data.json")
+        .then(data => res.send(data));
+});
+
+
+// The API contract is in a state of flux. This endpoint may be removed at some point.
+// This endpoint is similar to DELETE /api/v1/connections
+app.post("/api/v1/channel/leave", (req, res) => {
+    const reqPayload = req.body;
+
+    readFile("./connection-data.json")
+        .then(data => JSON.parse(data))
+        .then(data => {
+            const relation = data.items.find(rel => 
+                new String(rel.clientId).valueOf() === new String(reqPayload.clientId).valueOf()
+                &&
+                new String(rel.channelId).valueOf() === new String(reqPayload.channelId).valueOf()
+                );
+            
+            if (!relation) {
+                return new Promise(resolve => {
+                    res
+                        .status(422)
+                        .send({message: `Capability with id ${reqPayload.clientId} hasn't joined Channel with id ${reqPayload.channelId}`});
+                    resolve();
+                });
+            } else {
+                const desiredRelations = data.items.filter(rel => 
+                    rel.clientId.valueOf() === relation.clientId.valueOf() 
+                    ?
+                    rel.channelId.valueOf() !== relation.channelId.valueOf()
+                    :
+                    true         
+                    );
+                
+                data.items = desiredRelations;
+
+                return Promise.resolve(serialize(data))
+                    .then(json => writeFile("./connection-data.json", json))
+                    .then(() => console.log(`Capability with id ${reqPayload.clientId} has left Channel with id ${reqPayload.channelId}`))
+                    .then(() => res.sendStatus(200));
+            }
+        })
+        .catch(err => {
+            console.log("Error: " + err);
+            res.status(500).json(err);
+        });
+});
+
+// The API contract is in a state of flux. This endpoint may be removed at some point.
+// This endpoint is similar to POST /api/v1/connections
+app.post("/api/v1/channel/join", (req, res) => {

```

Comment:
This endpoint is outdated, delete the code if not needed

Reply:
On it!